src/reactviews/common/forms/formUtils.tsx

```diff
@@ -0,0 +1,228 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useContext, useEffect, useState } from "react";
+import { Input, Button, Textarea, Dropdown, Checkbox, Option, makeStyles, Field, InfoLabel, LabelProps } from "@fluentui/react-components";
+import { EyeRegular, EyeOffRegular } from "@fluentui/react-icons";
+
+import { ConnectionDialogContextProps, FormComponent, FormComponentType, IConnectionDialogProfile } from "../../../sharedInterfaces/connectionDialog";
+import { ConnectionDialogContext } from "../../pages/ConnectionDialog/connectionDialogStateProvider";
+
+export const FormInput = ({ value, target, type }: { value: string, target: keyof IConnectionDialogProfile, type: 'input' | 'password' | 'textarea' }) => {
+	const connectionDialogContext = useContext(ConnectionDialogContext);
+	const [inputVal, setValueVal] = useState(value);
+	const [showPassword, setShowPassword] = useState(false);
+
+	useEffect(() => {
+		setValueVal(value);

```

Comment:
nit: not the most intuitive method name/call.  i.e. this is passing the value that will be the val passed to setValue which will update some value.  I guess since this is generic method maybe that is as precise as we can be here?